transport/ws/src/main/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactory.java

```diff
@@ -82,15 +81,58 @@ public WebSocketExtensionSpi negotiate(ExtensionHeader extension, WsResourceAddr
      * @return object representing the list of negotiated  WebSocketExtensionSpi instances
      * @throws ProtocolException
      */
-    public ActiveExtensions negotiateWebSocketExtensions(ResourceAddress address,
-                                                         HttpAcceptSession session,
-                                                         String headerName,
-                                                         List<String> clientRequestedExtensions) throws ProtocolException {
-        // TODO: implement this method, based loosely on WsExtensionUtils.negotiateWebSocketExtensions (it will replace
-        // that method). Note that (as commented in that method) we do not need to have any notion of mandatory extensions 
-        // the client must negotiate.
-        throw new UnsupportedOperationException ();
-        //return ActiveExtensions.EMPTY;
+    public ActiveExtensions negotiateWebSocketExtensions(WsResourceAddress address, HttpAcceptSession session,
+        String headerName, List<String> clientRequestedExtensions) throws ProtocolException {
+
+        ActiveExtensions result = ActiveExtensions.EMPTY;
+        if (clientRequestedExtensions != null) {
+            // Maybe come back and cache supportedExtensionHeaders ??, but need more info on when they are loaded to
+            // decide
```

Comment:
Yeah, we should do this now. We can create supportedExtensionHeaders in the constructor since it is created from factoriesRO which is created in the constructor and immutable from then on.